producer/types.go

```diff
@@ -24,12 +24,12 @@ import (
 	"github.com/m3db/m3cluster/services"
 )
 
-// DataFinalizeReason defines the reason why the data is being finalized by Producer.
-type DataFinalizeReason int
+// FinalizeReason defines the reason why the data is being finalized by Producer.
+type FinalizeReason int
```

Comment:
Should there be a default value Unknown FinalizeReason? Thinking of a case where the default value is Consumed which may not be ideal for whatever reason.

Reply:
It does not go through any marshalling/unmarshalling so it's kinda safe to not have `Unknown` type